### PR TITLE
kata(-cc): upgrade to LSG release v2405.9.2

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

This pull request is for a version upgrade of the LSG release to v2405.9.2, involving updates to dependencies and version numbers in the project.

Important changes:
- Added cloud-hypervisor-cvm to core and updated to 38.0.72
- Updated cloud-hypervisor-cvm SPEC to make it match cloud-hypervisor as much as possible
- Update cloud-hypervisor-cvm SPEC to install where normal clh used to install and made it conflict with clh. We now only rely on cloud-hypervisor-cvm for kata(-cc)
- Updated kata-containers-cc softlinks in SPEC to use same cloud-hypervisor binary for both vanilla and snp config
- Upgraded other relevant LSG bits
- Upgraded kata(-cc) to new version compatible with latest cloud-hypervisor-cvm

###### Change Log  

- cloud-hypervisor-cvm
- cloud-hypervisor
- kata-containers-cc
- kata-containers
- kernel-mshv
- kernel-uvm

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- buddy-build (ensuring packages build): https://dev.azure.com/mariner-org/mariner/_build/results?buildId=578368&view=results
  - propietary build:  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=579323&view=results
- dev-image build(ensuring image builds with new packages): https://dev.azure.com/mariner-org/mariner/_build/results?buildId=579338&view=results
- test (ensuring image performs as expected): https://dev.azure.com/mariner-org/mariner/_build/results?buildId=579410&view=ms.vss-test-web.build-test-results-tab

### CODE REVIEW

**Comment by Redent0r** [comment]
Buddybuild: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=578368&view=results

**Comment by Redent0r** [comment]
PTAL at feedback in original PR https://github.com/microsoft/azurelinux/pull/9161/files#diff-e030ce8ff9f0cb3e493e5851321344b2377eb88823ea616c08b55fe6f3f56955

File: SPECS/kata-containers-cc/kata-containers-cc.spec [hunk 1]
```diff
@@ -289,6 +289,12 @@ install -D -m 0755 %{_builddir}/%{name}-%{version}/tools/osbuilder/image-builder
 %exclude %{osbuilder}/tools/osbuilder/rootfs-builder/ubuntu
 
 %changelog
+* Wed May 29 2024 CBL-Mariner Servicing Account <cblmargh@microsoft.com> - 3.2.0.azl2-1
+- Auto-upgrade to 3.2.0.azl2
+
+* Mon May 20 2024 Saul Paredes <saulparedes@microsoft.com> - 3.2.0.azl1-2
```
Comment (ms-mahuber) [inline SPECS/kata-containers-cc/kata-containers-cc.spec:295]: qq - usually one changelog entry per new PR. Is this intended? I do not see 3.2.0.azl1-2 to be on CBL-Mariner/main...

Comment (Redent0r) [reply SPECS/kata-containers-cc/kata-containers-cc.spec:295]: Good catch!

**Review by ms-mahuber — APPROVED** [review]

**Comment by Redent0r** [comment]
3.0 PR https://github.com/microsoft/azurelinux/pull/9313

File: SPECS/kata-containers-cc/kata-containers-cc.spec [hunk 2]
```diff
@@ -161,7 +161,7 @@ mkdir -p %{buildroot}/etc/systemd/system/containerd.service.d/
 # for testing policy/snapshotter without SEV SNP we use CH (with kernel-uvm and initrd) instead of CH-CVM with IGVM
 # Note: our kata-containers config toml expects cloud-hypervisor and kernel under a certain path/name, so we align this through symlinks here
```
Comment (christopherco) [inline SPECS/kata-containers-cc/kata-containers-cc.spec:162]: Now that we're pointing both symlinks to the same CLH binary, please also update this comment or remove the parts that aren't valid anymore

**Review by christopherco — APPROVED** [review]

**Pull request merged**